teps/00XX-interceptor-plugins.md

```diff
@@ -0,0 +1,389 @@
+ 
+2. Interceptors are configured or installed via an Interceptor CRD that describes an address for the Interceptor as well as  a list of valid parameters that the interceptor accepts.
+
+3. Built-in interceptors are still packaged as part of Triggers but run as separate processes instead of in the controller.
+
+4. Users configure interceptors in their Triggers and EventListeners just as they do today.
+
+5. For backwards compatibility, we will keep supporting the old WebhookInterceptor interface until beta. However, we will recommend users use the new Interceptor interface instead for writing any new Interceptors.
+
+See the [Design Details](#design-details) section below for more.
+
+
+### Notes/Constraints/Caveats
+
+1. Deployment Footprint and Scaling Concerns
+
+  The proposal suggests one interceptor deployment per cluster. This can cause scaling concerns if one Trigger/Listener takes away too much from others.The proposal also increases the deployment footprint for Triggers since now the 4 built-in interceptors are deployed as separate pods. In practice, these should be fairly small stateless services that are only created once per cluster. In addition, a HorizontalPodAutoscaler can be configured to scale the resources if needed. Finally, we could also consider bundling all 4 built-in interceptors into one service. 
+
+2. Logging/Debugging
+
+    Fetching logs from the Interceptor pods to debug a failed trigger is going to be cumbersome. So, each interceptor should return a descriptive errorMessage/status/summary in its response that will be logged by the EventListener as well as sent back as a response to the incoming event
+
+3. Backwards Compatibility
+
+    This model diverges from the existing WebHookInterceptor model. For backwards compatibility, we can keep supporting the existing [WebhookInterceptor interface](https://github.com/tektoncd/triggers/blob/master/docs/eventlisteners.md#event-interceptor-services) for a few releases or until beta.
+
+4. Secrets and RBAC concerns 
+
+    Today, if an interceptor needs access to a secret (e.g GitHub payload validation), the service account for the EventListener needs to get access to secrets. With the proposed model, this is no longer the case. Instead the GitHub interceptor deployment should use a role that has read access to secrets. This role would have to be a cluster role if it needs access to secrets across multiple namespaces. A security conscious admin might want to use resourceNames to lock down the list of secrets the deployment has access to.
```

Comment:
yeah resourceNames are a field in Role resource spec that allows you to restrict access to only certain named resources: https://kubernetes.io/docs/reference/access-authn-authz/rbac/#referring-to-resources

Added a link

(It does look like this might be less useful than I thought tho: https://github.com/kubernetes/kubernetes/issues/56582#issuecomment-411061150)